Server fails to load functions until /reload

When the server starts and has to load datapacks (with functions) I get a lot of these errors:
```
[Worker-Main-5/ERROR]: Failed to load function timber:settings/toggle/interfere
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "me.lucko.luckperms.common.verbose.VerboseHandler.offerPermissionCheckEvent(me.lucko.luckperms.common.verbose.event.PermissionCheckEvent$Origin, me.lucko.luckperms.common.verbose.VerboseCheckTarget, net.luckperms.api.query.QueryOptions, String, me.lucko.luckperms.common.calculator.result.TristateResult)" because the return value of "me.lucko.luckperms.fabric.LPFabricPlugin.getVerboseHandler()" is null
        at java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) [?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source) [?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinPool.scan(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) [?:?]
        at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "me.lucko.luckperms.common.verbose.VerboseHandler.offerPermissionCheckEvent(me.lucko.luckperms.common.verbose.event.PermissionCheckEvent$Origin, me.lucko.luckperms.common.verbose.VerboseCheckTarget, net.luckperms.api.query.QueryOptions, String, me.lucko.luckperms.common.calculator.result.TristateResult)" because the return value of "me.lucko.luckperms.fabric.LPFabricPlugin.getVerboseHandler()" is null
        at me.lucko.luckperms.fabric.listeners.PermissionCheckListener.onOtherPermissionCheck(PermissionCheckListener.java:86) ~[LuckPerms-Fabric-5.3.57.jar:?]
        at me.lucko.luckperms.fabric.listeners.PermissionCheckListener.onPermissionCheck(PermissionCheckListener.java:65) ~[LuckPerms-Fabric-5.3.57.jar:?]
        at me.lucko.fabric.api.permissions.v0.PermissionCheckEvent.lambda$static$0(PermissionCheckEvent.java:42) ~[55238d72-793f-40b3-8ff6-f2b3eda3ea46.jar:?]
        at me.lucko.fabric.api.permissions.v0.Permissions.getPermissionValue(Permissions.java:53) ~[55238d72-793f-40b3-8ff6-f2b3eda3ea46.jar:?]
        at me.lucko.fabric.api.permissions.v0.Permissions.check(Permissions.java:66) ~[55238d72-793f-40b3-8ff6-f2b3eda3ea46.jar:?]
        at com.github.tjeukayim.commandpermissionsfabric.PermissionsMod.lambda$alterCommand$2(PermissionsMod.java:53) ~[minecraft-command-permissions-1.3.0.jar:?]
        at com.mojang.brigadier.tree.CommandNode.canUse(CommandNode.java:65) ~[intermediary-server.jar:?]
        at com.mojang.brigadier.CommandDispatcher.parseNodes(CommandDispatcher.java:358) ~[intermediary-server.jar:?]
        at com.mojang.brigadier.CommandDispatcher.parse(CommandDispatcher.java:348) ~[intermediary-server.jar:?]
        at net.minecraft.class_2158.method_9195(class_2158.java:56) ~[intermediary-server.jar:?]
        at net.minecraft.class_5349.method_29451(class_5349.java:86) ~[intermediary-server.jar:?]
        ... 7 more
```

The affected functions dont load at all so all the datapacks that require them won't work.
However, after issuing a `/reload` command everything works fine again without any observable issues.

I am not sure if this is either an issue from LuckPerms or minecraft-command-permissions-fabric.

Loaded Mods:
```
- autoconfig1u@3.3.1
- fabric@0.37.2+1.17
- fabric-api-base@0.3.0+a02b446318
- fabric-api-lookup-api-v1@1.2.0+2b5c62d018
- fabric-biome-api-v1@3.1.11+c345aea818
- fabric-blockrenderlayer-v1@1.1.5+a02b446318
- fabric-command-api-v1@1.1.3+5ab9934c18
- fabric-commands-v0@0.2.2+92519afa18
- fabric-containers-v0@0.1.12+a02b446318
- fabric-content-registries-v0@0.2.4+2e8bd82f18
- fabric-crash-report-info-v1@0.1.5+be9da31018
- fabric-dimensions-v1@2.0.11+6cefd57718
- fabric-entity-events-v1@1.1.0+a02b446318
- fabric-events-interaction-v0@0.4.9+a722d8c018
- fabric-events-lifecycle-v0@0.2.1+92519afa18
- fabric-game-rule-api-v1@1.0.7+6cefd57718
- fabric-item-api-v1@1.2.4+a02b446318
- fabric-item-groups-v0@0.2.10+b7ab612118
- fabric-key-binding-api-v1@1.0.4+a02b446318
- fabric-keybindings-v0@0.2.2+36b77c3e18
- fabric-lifecycle-events-v1@1.4.4+a02b446318
- fabric-loot-tables-v1@1.0.4+a02b446318
- fabric-mining-levels-v0@0.1.3+92519afa18
- fabric-models-v0@0.3.0+a02b446318
- fabric-networking-api-v1@1.0.13+2e8bd82f18
- fabric-networking-blockentity-v0@0.2.11+a02b446318
- fabric-networking-v0@0.3.2+92519afa18
- fabric-object-builder-api-v1@1.10.9+b7ab612118
- fabric-object-builders-v0@0.7.3+a02b446318
- fabric-particles-v1@0.2.4+a02b446318
- fabric-permissions-api-v0@0.1-SNAPSHOT
- fabric-registry-sync-v0@0.7.10+e2961fee18
- fabric-renderer-api-v1@0.4.4+5f02c96918
- fabric-renderer-indigo@0.4.8+a02b446318
- fabric-renderer-registries-v1@3.2.0+a02b446318
- fabric-rendering-data-attachment-v1@0.1.5+a02b446318
- fabric-rendering-fluids-v1@0.1.13+a02b446318
- fabric-rendering-v0@1.1.2+92519afa18
- fabric-rendering-v1@1.7.0+2e8bd82f18
- fabric-resource-loader-v0@0.4.8+a00e834b18
- fabric-screen-api-v1@1.0.4+155f865c18
- fabric-screen-handler-api-v1@1.1.8+a02b446318
- fabric-structure-api-v1@1.1.13+5ab9934c18
- fabric-tag-extensions-v0@1.1.4+a02b446318
- fabric-textures-v0@1.0.6+a02b446318
- fabric-tool-attribute-api-v1@1.2.12+b7ab612118
- fabric-transfer-api-v1@1.0.0+3fe3c3f518
- fabricloader@0.11.6
- fabricproxy-lite@1.1.5
- java@16
- luckperms@5.3.57
- minecraft@1.17.1
- minecraft-command-permissions@1.3.0
```

[Complete Server Log](https://github.com/TjeuKayim/minecraft-command-permissions-fabric/files/6965473/full-log.txt)

--- a/gradle.properties
+++ b/gradle.properties
@@ -8,7 +8,7 @@ org.gradle.jvmargs=-Xmx1G
 	loader_version=0.11.3
 
 # Mod Properties
-	mod_version = 1.3.0
+	mod_version = 1.4.0
 	maven_group = com.github.tjeukayim
 	archives_base_name = minecraft-command-permissions
 

--- a/src/main/java/com/github/tjeukayim/commandpermissionsfabric/PermissionsMod.java
+++ b/src/main/java/com/github/tjeukayim/commandpermissionsfabric/PermissionsMod.java
@@ -21,7 +21,8 @@ public class PermissionsMod implements ModInitializer {
     @Override
     public void onInitialize() {
         // LuckPerms only calls AbstractLuckPermsPlugin::enable on the SERVER_STARTING event
-        // before that LuckPermsPlugin::getVerboseHandler() returns null
+        // before that LuckPermsPlugin::getVerboseHandler() returns null. Initializing
+        // earlier caused issued when loading datapacks with functions.
         ServerLifecycleEvents.SERVER_STARTING.register(server -> {
             var dispatcher = server.getCommandManager().getDispatcher();
             if ("true".equals(System.getenv("minecraft-command-permissions.test"))) {

--- a/src/main/java/com/github/tjeukayim/commandpermissionsfabric/PermissionsMod.java
+++ b/src/main/java/com/github/tjeukayim/commandpermissionsfabric/PermissionsMod.java
@@ -1,10 +1,9 @@
 package com.github.tjeukayim.commandpermissionsfabric;
 
-import com.mojang.brigadier.CommandDispatcher;
 import com.mojang.brigadier.tree.CommandNode;
 import me.lucko.fabric.api.permissions.v0.Permissions;
 import net.fabricmc.api.ModInitializer;
-import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
 import net.minecraft.server.command.ServerCommandSource;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
@@ -21,7 +20,10 @@ public class PermissionsMod implements ModInitializer {
 
     @Override
     public void onInitialize() {
-        CommandRegistrationCallback.EVENT.register((dispatcher, dedicated) -> {
+        // LuckPerms only calls AbstractLuckPermsPlugin::enable on the SERVER_STARTING event
+        // before that LuckPermsPlugin::getVerboseHandler() returns null
+        ServerLifecycleEvents.SERVER_STARTING.register(server -> {
+            var dispatcher = server.getCommandManager().getDispatcher();
             if ("true".equals(System.getenv("minecraft-command-permissions.test"))) {
                 var allCommands = dispatcher.getRoot().getChildren()
                         .stream()

